components/service/glean/src/main/java/mozilla/components/service/glean/storages/UuidsStorageEngine.kt

```diff
@@ -49,16 +89,76 @@ internal object UuidsStorageEngine {
      * @return the uuids recorded in the requested store
      */
     @Synchronized
-    public fun getSnapshot(storeName: String, clearStore: Boolean): MutableMap<String, UUID>? {
+    fun getSnapshot(storeName: String, clearStore: Boolean): UUIDStorage? {
+        val allLifetimes: UUIDStorage = mutableMapOf()
+
+        // Make sure data with "user" lifetime is loaded before getting the snapshot.
+        userLifetimeStorage.all
+
+        // Get the metrics for all the supported lifetimes.
+        for ((_, store) in uuidStores) {
+            store[storeName]?.let {
+                allLifetimes.putAll(it)
+            }
+        }
+
         if (clearStore) {
-            return uuidStores.remove(storeName)
+            // We only allow clearing metrics with the "ping" lifetime.
+            uuidStores[Lifetime.Ping.toString()]!!.remove(storeName)
         }
 
-        return uuidStores.get(storeName)
+        return if (allLifetimes.isNotEmpty()) allLifetimes else null
+    }
+
+    /**
+     * Get a snapshot of the stored data as a JSON object.
+     *
+     * @param storeName the name of the desired store
+     * @param clearStore whether or not to clearStore the requested store
```

Comment:
```suggestion
     * @param clearStore whether or not to clear the requested store
```

Reply:
Should this doc contain that it only clears metrics with a Ping lifetime?

Reply:
Good point. I've moved this to `GenericScalarStorageEngine` and made it clearer.